test/server/core/card/CardHighlighting.spec.ts

```diff
@@ -0,0 +1,67 @@
+describe('Card highlighting', function() {
+    integration(function(contextRef) {
+        describe('Card highlighting', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        resources: ['armed-to-the-teeth',
+                            'collections-starhopper',
+                            'covert-strength',
+                            'chewbacca#pykesbane',
+                            'battlefield-marine',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                            'moment-of-peace',
+                        ],
+                        hand: ['vanguard-infantry'],
+                        groundArena: ['wampa'],
+                        spaceArena: ['cartel-spacer'],
+                    },
+                    player2: {
+                        groundArena: ['death-trooper'],
+                        spaceArena: ['mercenary-gunship']
+                    }
+                });
+            });
+
+            it('the prompt before an action and after should be different.', function () {
+                const { context } = contextRef;
+                let currentPossibleActions = context.player1.currentPrompt();
+
+                const reset = () => {
+                    currentPossibleActions = context.player1.currentPrompt();
+                };
+
+                // attack action
+                context.player1.clickCard(context.wampa);
+                context.player1.clickCard(context.deathTrooper);
+                context.player2.claimInitiative();
+                expect(context.player1.currentPrompt()).not.toEqual(currentPossibleActions);
+
+                reset();
+
+                // smuggle action
+                context.player1.clickCard(context.collectionsStarhopper);
+                expect(context.player1.currentPrompt()).not.toEqual(currentPossibleActions);
+
+                reset();
+
+                // play from hand action
+                context.player1.clickCard(context.vanguardInfantry);
```

Comment:
I thought we discussed that you were having some issue with clicks not working right because something about making the cards selectable broke the test harness, what ended up happening with that?

Reply:
Well i haven't run into that issue anymore ever since I refactored how clickCards worked. I must have removed some logic and returned it back which fixed the issue.